bot/exts/halloween/halloweenify.py

```diff
@@ -37,11 +38,25 @@ async def halloweenify(self, ctx: commands.Context) -> None:
                 f"**{ctx.author.display_name}** wasn\'t spooky enough for you? That\'s understandable, "
                 f"{ctx.author.display_name} isn\'t scary at all! "
                 "Let me think of something better. Hmm... I got it!\n\n "
-                f"Your new nickname will be: \n :ghost: **{nickname}** :jack_o_lantern:"
             )
             embed.set_image(url=image)
 
-            await ctx.author.edit(nick=nickname)
+            try:
+                if isinstance(ctx.author, discord.Member):
+                    await ctx.author.edit(nick=nickname)
+                    embed.description += f"Your new nickname will be: \n:ghost: **{nickname}** :jack_o_lantern:"
+
+                else:   # The command has been invoked in DM
+                    embed.description += (
+                        f"Your new nickname should be: \n :ghost: **{nickname}** :jack_o_lantern: \n\n"
+                        f"Feel free to change it yourself, or invoke the command again inside the server."
+                    )
```

Comment:
I'd prefer if only the line that we expect to raise was in the `try` block, e.g.:
```python
if isinstance(...):
    try:
        await ctx.author.edit(...)
    except Forbidden:
        # append no permissions message
    else:
        # append success message
else:
    # append DM message
```
This way the logic for both cases (guild / DM) is clearly separated, and we see exactly which lines may raise.

I like the comment on line 49 - helps see why we're branching there.